runtime/lua/vim/shared.lua

```diff
@@ -983,4 +983,22 @@ do
   end
 end
 
+--- @generic T
+--- @param root string
+--- @param _mod T
+--- @return T
+function vim._deferload_submodules(root, _mod)
```

Comment:
Is "submodules" telling the reader anything useful? Including "require" in the name may be more intuitive.

```suggestion
function vim._defer_require(root, _mod)
```